### PR TITLE
Adding action server to allow scheduling of fremen model construction

### PR DESCRIPTION
**Pull request by Jailander**
I am adding an action server to build the fremen models for topological prediction.
Previously these models were only updated when the navigation was launched, so now it is necessary to rebuild these model periodically for long term operation.
Also even though the prediction itself is a service, the building of the models has been made an action server so it can be scheduled. The prediction will remain a service so no changes in other's code are necessary.
This PR does this in  topological_navigation/scripts/topological_prediction.py the changes in other files are necessary for the creation of the action itself.

### CODE REVIEW

**Comment by Jailander** [comment]
Also I have added the total duration of the construction action in the feedback so it can be scheduled accordingly but for references in a medium/large map with 179 edges it took always less than one minute

**Comment by bfalacerda** [comment]
looks good, thanks. I'll add this to bob and add it as a night task. I'll let you know how it works

File: topological_navigation/scripts/topological_prediction.py [hunk 1]
```diff
@@ -79,18 +63,21 @@ def __init__(self, name) :
         self.FremenClient.wait_for_server()
         rospy.loginfo(" ...done")
 
-        self.predict_srv=rospy.Service('/topological_prediction/predict_edges', strands_navigation_msgs.srv.PredictEdgeState, self.predict_edge_cb)
-        #This service returns a list of nodes that have a given tag
+
+        #Creating Action Server
+        rospy.loginfo("Creating action server.")
+        self._as = actionlib.SimpleActionServer(name, strands_navigation_msgs.msg.BuildTopPredictionAction, execute_cb = self.BuildCallback, auto_start = False)
```
Comment (bfalacerda) [inline topological_navigation/scripts/topological_prediction.py]: it's weird using the node name here, as this is the action to build a model. Why did you do it like this?

Comment (Jailander) [inline topological_navigation/scripts/topological_prediction.py]: True I didn't think about it that is why I can Chang it easily sorry

Comment (Jailander) [inline topological_navigation/scripts/topological_prediction.py]: @bfalacerda should I change this, or is it too late now?

Comment (bfalacerda) [inline topological_navigation/scripts/topological_prediction.py]: please change

**Comment by Jailander** [comment]
the new action name is `/topological_prediction/build_temporal_model`

**Comment by marc-hanheide** [comment]
Let me know when this has been confirmed working, @bfalacerda

**Comment by bfalacerda** [comment]
works. merging.

**Pull request merged**